exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,259 +648,559 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
+			}
+			pair = o[i].Pair
+		} else if o[i].AssetType != asset.Spot {
+			return nil, fmt.Errorf("%w: %v", asset.ErrNotSupported, o[i].AssetType)
+		}
+		keyInfo := assetAndType{o[i].AssetType, o[i].Type, pair}
+		if assetAndTypeToIDsMap[keyInfo] == nil {
+			assetAndTypeToIDsMap[keyInfo] = &orderAndClientSuppliedIDs{
+				orderIDs:       []string{},
+				clientOrderIDs: []string{},
+			}
+		}
 		switch {
-		case o[i].ClientOrderID != "":
-			clientOrderIDs = append(clientOrderIDs, o[i].ClientOrderID)
 		case o[i].OrderID != "":
-			orderIDs = append(orderIDs, o[i].OrderID)
+			assetAndTypeToIDsMap[keyInfo].orderIDs = append(assetAndTypeToIDsMap[keyInfo].orderIDs, o[i].OrderID)
+		case o[i].ClientOrderID != "":
+			assetAndTypeToIDsMap[keyInfo].clientOrderIDs = append(assetAndTypeToIDsMap[keyInfo].clientOrderIDs, o[i].ClientOrderID)
 		default:
 			return nil, order.ErrOrderIDNotSet
 		}
 	}
-	cancelledOrders, err := e.CancelMultipleOrdersByIDs(ctx, orderIDs, clientOrderIDs)
-	if err != nil {
-		return nil, err
-	}
 	resp := &order.CancelBatchResponse{
 		Status: make(map[string]string),
 	}
-	for i := range cancelledOrders {
-		if cancelledOrders[i].ClientOrderID != "" {
-			resp.Status[cancelledOrders[i].ClientOrderID] = cancelledOrders[i].State + " " + cancelledOrders[i].Message
-			continue
+	for k, v := range assetAndTypeToIDsMap {
+		if k.aType == asset.Spot {
+			switch k.oType {
+			case order.Market, order.Limit, order.AnyType, order.UnknownType:
+				cancelledOrders, err := e.CancelOrdersByIDs(ctx, v.orderIDs, v.clientOrderIDs)
+				if err != nil {
+					return nil, err
+				}
+				for _, co := range cancelledOrders {
+					if slices.Contains(v.orderIDs, co.OrderID) {
+						resp.Status[co.OrderID] = co.State + " " + co.Message
+					} else {
+						resp.Status[co.ClientOrderID] = co.State + " " + co.Message
+					}
+				}
+			case order.Stop, order.StopLimit, order.TrailingStop, order.TrailingStopLimit:
+				cancelledOrders, err := e.CancelMultipleSmartOrders(ctx, &CancelOrdersRequest{
+					OrderIDs:       v.orderIDs,
+					ClientOrderIDs: v.clientOrderIDs,
+				})
+				if err != nil {
+					return nil, err
+				}
+				for _, co := range cancelledOrders {
+					if slices.Contains(v.orderIDs, co.OrderID) {
+						resp.Status[co.OrderID] = co.State + " " + co.Message
+					} else {
+						resp.Status[co.ClientOrderID] = co.State + " " + co.Message
+					}
+				}
+			default:
+				return nil, fmt.Errorf("%w: %s", order.ErrUnsupportedOrderType, k.oType.String())
+			}
+		} else {
+			cancelledOrders, err := e.CancelMultipleFuturesOrders(ctx, &CancelFuturesOrdersRequest{
+				Symbol:         k.pair,
+				OrderIDs:       v.orderIDs,
+				ClientOrderIDs: v.clientOrderIDs,
+			})
+			if err != nil {
+				return nil, err
+			}
+			for _, co := range cancelledOrders {
+				cancellationStatus := order.Cancelled.String()
+				if co.Code != 200 && co.Code != 0 {
+					cancellationStatus = "Failed"
+				}
```

Comment:
🚧 Change request:
I've been accepting/ignoring this `Code == 0` is alright paradigm everywhere, and just realised to question it.
Why are you saying that a code of 0 is okay?
The docs don't support that.